app/Http/Controllers/VendorController.php

```diff
@@ -2,110 +2,55 @@
+    // Vendor logout ** REFACTORED **
+    public function logout(NotificationService $notification): RedirectResponse 
     {
-        //Logout the user
         Auth::guard('web')->logout();
+        session()->invalidate();
+        session()->regenerateToken();
 
-        //Invalidate the session
-        $request->session()->invalidate();
-
-        //Regenerate the CSRF token
-        $request->session()->regenerateToken();
-
-        //Redirect to the vendor login page
-        return redirect('/vendor/login');
+        return redirect('/vendor/login')->with($notification->message('You have been logged out successfully', 'success'));
     }
 
-    // Vendor Profile
-    public function vendorProfile()
-    {
-        $id = Auth::user()->id;
-        $vendorProfile = User::find($id);
-        return view('vendor.profile.vendor_profile', compact('vendorProfile'));
+    // Vendor Profile  ** REFACTORED **
+    public function vendorProfile(User $user)
+    {   
+        return view('vendor.profile.vendor_profile', ['user' => $user]);
     }
 
-    // Vedor Profile Store
-    public function vendorProfileStore(Request $request)
-    {
-        // Get the authenticated user's ID
-        $id = Auth::user()->id;
-
-        // Find the user in the database using the retrieved ID
-        $data = User::find($id);
-
-        // Update user's profile information with data from the request
-        $data->firstname = $request->firstname;
-        $data->lastname = $request->lastname;
-        $data->username = $request->username;
-        $data->email = $request->email;
-        $data->github = $request->github;
-        $data->instagram = $request->instagram;
-        $data->linkedin = $request->linkedin;
-        $data->job_title = $request->job_title;
-        $data->phone = $request->phone;
-        $data->address = $request->address;
-        $data->postcode = $request->postcode;
-        $data->vendor_shop_name = $request->vendor_shop_name;
-        $data->vendor_join = $request->vendor_join;
-        $data->vendor_short_info = $request->vendor_short_info;
-
-        // Check if a new profile photo was uploaded
-        if ($request->file('photo')) {
-            $file = $request->file('photo');
-
-            // Unlink the images
-            @unlink(public_path('upload/vendor_profile_image/'. $data->photo));
-            
-            // Generate a unique filename for the uploaded photo
-            $filename = date('Y-m-d H:i:s') . $file->getClientOriginalName();
-            
-            // Move the uploaded photo to the specified directory
-            $file->move(public_path('upload/vendor_profile_image'), $filename);
-            
-            // Update the 'photo' field in the user's data with the new filename
-            $data->photo = $filename;
-        }
-
-        // Save the updated user data to the database
-        $data->save();
+    // Vedor Profile Update ** REFACTORED **
```

Comment:
This part seems to have an error, Too few arguments to function App\Http\Controllers\VendorController::update(), 3 passed in /Users/natalisuportman/mecom/vendor/laravel/framework/src/Illuminate/Routing/Controller.php on line 54 and exactly 4 expected

I don't know where the issue is because the $file is not passed as dependency Injection and why returning 3 parameters while expecting 4. I have seen the UploadedFile from Laravel and I used that but it seems can't resolve the path string

Reply:
You should remove the `$file` from the arguments. It cannot be resolved by the container as it is neither a service or passed in the request like that.
Also try to keep dependencies and arguments resolved from the router in an order. So in  this case the `User` should be the last argument.